### DISCUSSION TITLE
Is it possible to use vulkan?

### DISCUSSION THREAD

**Discussion by Raj2032** [fltk-rs/fltk-rs#1110] [General]
Over at `fltk`, I am able to switch over to vulkan as seen in this post:

https://www.fltk.org/str.php?L3372

So I was wondering with `fltk-rs` if it was possible to also use vulkan over opengl, and if so would it involve a lot more work?

Would the code also be the same if using `fltk-rs` with vulkan?

**Comment by MoAlyousef** [comment]
You can use vulkan via wgpu-rs to draw on an fltk window:

https://github.com/parasyte/pixels/tree/main/examples/minimal-fltk

https://github.com/fltk-rs/demos/tree/master/wgpu

But you can’t make FLTK use vulkan as a backend (called driver) for its widget drawing.

**Reply by Raj2032** [reply]
@MoAlyousef  I see thanks,

Another question is fltk-rs by any chance "backend agnostic"?

**Reply by MoAlyousef** [reply]
It’s not backend agnostic unfortunately. It uses specific backends depending on the operating system. MacOS -> Cocoa, Windows -> win32/gdi/gdiplus, other unixy systems -> libX11 (with ongoing work to support wayland natively (not via xwayland)).

**Reply by Raj2032** [reply]
Right I see mate, thanks

**Reply by MoAlyousef** [reply]
No problem.
